router/customdestinationmanager/customdestinationmanager_test.go

```diff
@@ -197,3 +269,154 @@ func TestKVManagerInvocations(t *testing.T) {
+	someDestination := backendconfig.DestinationT{
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	transformedResponse := transformedResponseJSON{
+		Message: map[string]interface{}{
+			"key":  "user:myuser-id",
+			"path": "mode-1",
+			"value": map[string]interface{}{
+				"key": "someKey",
+				"fields": map[string]interface{}{
+					"field1": "value1",
+					"field2": 2,
+				},
+			},
+		},
+		UserId: "myuser-id",
+	}
+
+	t.Run("When JSON module is loaded into Redis, path is mentioned and key is not present in Redis, should set into redis and fetching the data should be successful", func(t *testing.T) {
+		redisAddr, destroy := testutils.StartRedis(t, "redis/redis-stack-server", "latest")
+		defer destroy()
+		event, err := json.Marshal(transformedResponse)
+		if err != nil {
+			t.Errorf("MarshalError: %v\n", err.Error())
+			return
+		}
+		config := map[string]interface{}{
+			"useJSONModule": true,
+			"address":       redisAddr,
+			"db":            0,
+			"clusterMode":   false,
+		}
+		kvMgr := &kvstoremanager.RedisManagerT{
+			Config: config,
+		}
+		kvMgr.Connect()
+		db := kvMgr.GetClient()
+
+		stCd, _ := customManager.send(event, kvMgr, config)
+		v, err := db.JSONGet(context.TODO(), "user:myuser-id", "$.mode-1").Result()
+
+		require.Nil(t, err)
+		require.NotEqual(t, "", v)
+
+		require.JSONEq(t, `[{"key":"someKey","fields":{"field1":"value1","field2":2}}]`, v)
+
+		require.Equal(t, http.StatusOK, stCd)
+	})
+
+	t.Run("When JSON module is loaded into Redis, path is mentioned and key is present in Redis, should set into redis and fetching the data should be successful", func(t *testing.T) {
+		redisAddr, destroy := testutils.StartRedis(t, "redis/redis-stack-server", "latest")
+		defer destroy()
+		event, err := json.Marshal(transformedResponse)
+		if err != nil {
+			t.Errorf("MarshalError: %v\n", err.Error())
+			return
+		}
+		config := map[string]interface{}{
+			"useJSONModule": true,
+			"address":       redisAddr,
+			"db":            0,
+			"clusterMode":   false,
+		}
+		kvMgr := &kvstoremanager.RedisManagerT{
+			Config: config,
+		}
+		kvMgr.Connect()
+		db := kvMgr.GetClient()
+		ctx := context.TODO()
+
+		_, e := db.JSONSet(ctx, "user:myuser-id", "$", `{"mode-in":{"a":1,"size":"LM"}}`).Result()
+		require.Nil(t, e)
+
+		stCd, _ := customManager.send(event, kvMgr, config)
+		firstVal, err := db.JSONGet(ctx, "user:myuser-id", "$.mode-in").Result()
+		require.Nil(t, err)
+		require.NotEqual(t, "", firstVal)
+
+		v, err2 := db.JSONGet(ctx, "user:myuser-id", "$.mode-1").Result()
+		require.Nil(t, err2)
+		require.NotEqual(t, "", v)
+
+		require.JSONEq(t, `[{"a":1,"size":"LM"}]`, firstVal)
+		require.JSONEq(t, `[{"key":"someKey","fields":{"field1":"value1","field2":2}}]`, v)
+
+		require.Equal(t, http.StatusOK, stCd)
```

Comment:
We should keep the assertions where it is used,
```suggestion
		stCd, _ := customManager.send(event, kvMgr, c)
		require.Equal(t, http.StatusOK, stCd)

		firstVal, err := db.JSONGet(ctx, "user:myuser-id", "$.mode-in").Result()
		require.NoError(t, err)
		require.JSONEq(t, `[{"a":1,"size":"LM"}]`, firstVal)

		v, err := db.JSONGet(ctx, "user:myuser-id", "$.mode-1").Result()
		require.NoError(t, err)
		require.JSONEq(t, `[{"key":"someKey","fields":{"field1":"value1","field2":2}}]`, v)
```

Reply:
Done